src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service.documentremoval;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Iterator;
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DocumentRemovalService {
+
+    public void retrieveDocumentNodes(JsonNode root, List<JsonNode> documentNodes) {
+        if (root.isObject()) {
+            Iterator<String> fieldNames = root.fieldNames();
+            while (fieldNames.hasNext()) {
+                String fieldName = fieldNames.next();
+                JsonNode fieldValue = root.get(fieldName);
+                if (fieldValue.has("document_url")) {
+                    documentNodes.add(fieldValue);
+                } else {
+                    retrieveDocumentNodes(fieldValue, documentNodes);
+                }
+            }
+        } else if (root.isArray()) {
+            ArrayNode arrayNode = (ArrayNode) root;
+            for (int i = 0; i < arrayNode.size(); i++) {
+                JsonNode arrayElement = arrayNode.get(i);
+                retrieveDocumentNodes(arrayElement, documentNodes);
+            }
+        }
+    }
+}
```

Comment:
This is fine.  Someone might suggest moving "document_url" into a constant in this class.  Then the handlers can use it too.

I have done this on my later work though - so it's coming.